src/lux.jl

```diff
@@ -43,16 +56,14 @@ initialstates(rng::AbstractRNG, l::PolyLuxLayer) = _init_luxstate(rng, l.basis)
 
 (l::PolyLuxLayer)(args...) = evaluate(l, args...)
 
-function evaluate(l::PolyLuxLayer, x::SINGLE, ps, st)
-   B = acquire!(st.cache, :B, (length(l.basis), ), _valtype(l.basis, x))
-   evaluate!(parent(B), l.basis, x)
-   return B 
+function evaluate(l::PolyLuxLayer, X, ps, st)
+   
+   # TODO: after we make sure we want to migrate to HyperDualNumbers in any cases we can ignore_derivatives from ChainRulesCore
+   #B = ChainRulesCore.ignore_derivatives() do 
+   #   evaluate(l.basis, X)
+   #end
+   B = evaluate(l.basis, X)
```

Comment:
why do we no longer `ignore_derivatives`? I think most  bases don't have parameters and ignoring those in the back-propagation will significantly speed up the code? We may need two Lux Layers, mayb `P4MLLayer` and ConstP4MLLayer` ??

Reply:
When I use Zygote pullback to calculate the gradient w.r.t `X`, if I use ignore_derivatives the gradient returns nothing. I didn't go into details but I will look into it and try to keep ignore_derivatives.

Reply:
Ok - I understand. This is potentially a problem. We need to discuss this

Reply:
actually scratch my comment below, we need to sort this out as well.

Reply:
I pushed a double-pullback example to `temp/double_pb.jl` - this suggests that my concern was entirely unfounded. Maybe we can pull this off with minimal additional effort.

Reply:
Before we close this comment, we need to discuss the `ignore_derivatives` maybe part of the meeting tomorrow?